### ISSUE TITLE
Visualizing the Knowledge Gradient acquisition function

### ISSUE THREAD

**Issue by r-ashwin** [pytorch/botorch#580]
## Issue description
I am trying to visualize the KG acquisition function for simple 1D and 2D functions. I need this to analyze the acquisition in comparison with other myopic acquisition functions such as the EI. I use the `forward()` method of the `qKnowledgeGradient `class to compute it on a grid, but I don't see expected results. What I expect is a smooth curve with occasional discontinuities, e.g., in 

```
@article{frazier2009knowledge,
  title={The knowledge-gradient policy for correlated normal beliefs},
  author={Frazier, Peter and Powell, Warren and Dayanik, Savas},
  journal={INFORMS journal on Computing},
  volume={21},
  number={4},
  pages={599--613},
  year={2009},
  publisher={INFORMS}
}
```

**Question:** Am I right in passing `X_actual `with the first row to be `q=1` query point and the rest to be `num_fantasies `fantasy points?

## Code example
Minimal code example is as follows
```
import math
import torch

from botorch.fit import fit_gpytorch_model
from botorch.models import SingleTaskGP
from botorch.utils import standardize
from gpytorch.mlls import ExactMarginalLogLikelihood
```

```
bounds  = torch.stack([torch.zeros(1), torch.ones(1)])

train_X = bounds[0] + (bounds[1] - bounds[0]) * torch.rand(20, 1)
train_Y = torch.sin(2 * math.pi * train_X[:, [0]]) 
train_Y = standardize(train_Y + 0.05 * torch.randn_like(train_Y))

model   = SingleTaskGP(train_X, train_Y)
mll     = ExactMarginalLogLikelihood(model.likelihood, model)
fit_gpytorch_model(mll);
```
`true function`
![true_func](https://user-images.githubusercontent.com/11855653/97032465-7f627980-1527-11eb-9505-5ffb1ec64f08.png)
```
from botorch.acquisition import qKnowledgeGradient
torch.manual_seed(123)
num_fantasies = 2500
qKG           = qKnowledgeGradient(model, num_fantasies=num_fantasies)
aKG           = torch.zeros(test_X.shape[0])
for i in range(test_X.shape[0]):
    X_actual = torch.cat([test_X[i,:].reshape(-1,1),
                          torch.rand(num_fantasies,1)], dim=0)
    aKG[i]   = qKG.forward(X_actual).item()

```

```
# plot the acquisition function
plt.plot(test_X, aKG)
```
![aKG](https://user-images.githubusercontent.com/11855653/97032463-7ec9e300-1527-11eb-8382-449637b90ac9.png)

**Comment by Balandat** [comment]
So the value of the fantasy points is crucial here, you'll actually need to optimize those. Fortunately other smart people have thought about this and implemented a solution for this: https://github.com/pytorch/botorch/pull/515.
